src/System.Linq.Expressions/tests/BinaryOperators/GeneralBinaryTests.cs

```diff
@@ -205,5 +249,22 @@ public static bool IsNaN(object obj)
                 type, Expression.Constant(lhs), Expression.Constant(rhs), false, null, sillyLambda);
             Expression.Lambda(op).Compile(useInterpreter).DynamicInvoke();
         }
+
+        private class GenericClassWithNonGenericMethod<TClassType>
+        {
+            public static int DoIntStuff(int x, int y) => unchecked(x + y);
+        }
+
+        [Theory, MemberData(nameof(MethodAllowedBinaryTypesAndValues))]
+        public static void MethodOfOpenGeneric(ExpressionType type, Type lhs, object rhs)
+        {
+            ParameterExpression left = Expression.Parameter(lhs);
+            ConstantExpression right = Expression.Constant(rhs);
+            var genType = typeof(GenericClassWithNonGenericMethod<>);
+            MethodInfo method = genType.GetMethod(nameof(GenericClassWithNonGenericMethod<int>.DoIntStuff));
+            Assert.Throws<ArgumentException>("method", () => Expression.MakeBinary(type, left, right, false, method));
+            method = genType.MakeGenericType(genType).GetMethod(nameof(GenericClassWithNonGenericMethod<int>.DoIntStuff));
+            Assert.Throws<ArgumentException>("method", () => Expression.MakeBinary(type, left, right, false, method));
```

Comment:
Can/should this test that it then succeeds when it's made closed? That would help to confirm the cause of the failure being validated.